crates/gas/src/pricing/resolver.rs

```diff
@@ -2,7 +2,7 @@ use crate::Op;
 
 /// Pricing an `Op` or an import function should be part of the consensus protocol.
 /// (It's crucial since otherwise, different network peers will give the same function a different price).
-pub trait PriceResolver {
+pub trait PriceResolver: Clone {
```

Comment:
I didn't want to, but I need this in order to wrap it in a `ProgramPricing` struct when calling `ProgramPricing::new()`. Any ideas?

Reply:
It seems we'll have to substitute the `R` of `ProgramPricing` in favor of `Box<dyn PriceResolver>`

Reply:
I don't see how that will solve the `Clone` issue, am I missing something?

Reply:
Does it help?

Reply:
Also, I guess we could do as follows:

```rust
impl<R: PriceResolver + 'static> PriceResolver for &R {
}
```

Reply:
What do you mean? How should we instantiate a `ProgramPricing` given a `Box<dyn PriceResolver>` without the `Clone` bound?

Reply:
Actually, I think that the `PriceResolver` trait should eliminate the usage of `&self`.
The opcode price should behave like a constant...

Reply:
Ok, nevermind. See https://github.com/spacemeshos/svm/pull/265/commits/7b852245ddb23458a6934c7ff355a2d31e72ebbe. I moved the type bound on a `where` for consistency with the codebase and removed `'static` because I don't think it's needed. Is this fine?

Reply:
Possibly, but it would be nice to let the trait support a `PriceResolver` that can say, import opcode pricing from a file and stuff like that. I think we should leave it as it is